Makefile

```diff
@@ -1,111 +1,150 @@
-REPO := quay.io/boson/func
-BIN  := func
+# ##
+#
+# Run 'make help' for a summary
+#
+# ##
+
+# Binaries
+BIN         := func
+BIN_DARWIN  ?= $(BIN)_darwin_amd64
+BIN_LINUX   ?= $(BIN)_linux_amd64
+BIN_WINDOWS ?= $(BIN)_windows_amd64.exe
+
+# Version
+# A verbose version is built into the binary including a date stamp, git commit
+# hash and the version tag of the current commit (semver) if it exists.
+# If the current commit does not have a semver tag, 'tip' is used.
+DATE    := $(shell date -u +"%Y%m%dT%H%M%SZ")
+HASH    := $(shell git rev-parse --short HEAD 2>/dev/null)
+VTAG    := $(shell git tag --points-at HEAD)
+VERS    ?= $(shell [ -z $(VTAG) ] && echo 'tip' || echo $(VTAG) )
+LDFLAGS := "-X main.date=$(DATE) -X main.vers=$(VERS) -X main.hash=$(HASH)"
+
+# Templates
+# Built into the binary are the contents of ./templates.  This is done by
+# running 'pkger' which generates pkged.go, containing a go-encoded version of
+# the templates directory.
+PKGER           ?= pkger
+
+# Code is all go source files, used for build target freshness checks
+CODE := $(shell find . -name '*.go')
 
-PKGER?=pkger
+all: build
+	# Run 'make help' for make target documentation.
 
-DARWIN=$(BIN)_darwin_amd64
-LINUX=$(BIN)_linux_amd64
-WINDOWS=$(BIN)_windows_amd64.exe
+# Print Help Text
+help:
+	@echo 'Usage: make <OPTIONS> ... <TARGETS>'
+	@echo ''
+	@echo 'Available targets are:'
+	@awk 'BEGIN {FS = ":.*##"; printf "\nUsage:\n  make \033[36m<target>\033[0m\n"} /^[a-zA-Z0-9_-]+:.*?##/ { printf "  \033[36m%-15s\033[0m %s\n", $$1, $$2 } /^##@/ { printf "\n\033[1m%s\033[0m\n", substr($$0, 5) } ' $(MAKEFILE_LIST)
 
-CODE := $(shell find . -name '*.go')
-DATE := $(shell date -u +"%Y%m%dT%H%M%SZ")
-HASH := $(shell git rev-parse --short HEAD 2>/dev/null)
-VTAG := $(shell git tag --points-at HEAD)
-# a VERS environment variable takes precedence over git tags
-# and is necessary with release-please-action which tags asynchronously
-# unless explicitly, synchronously tagging as is done in ci.yaml
-VERS ?= $(shell [ -z $(VTAG) ] && echo 'tip' || echo $(VTAG) )
-
-TEMPLATE_DIRS=$(shell find templates -type d)
-TEMPLATE_FILES=$(shell find templates -type f -name '*')
```

Comment:
@matejvasek I removed these conditions because the target was always running (I think because we remove caches before each run), and I was getting:
`make: *** No rule to make target `templates/python/http/__pycache__', needed by `pkged.go'.  Stop.git` when running after unit testing of templates was performed.  Please let me know if I am missing something imprtant.